crates/core/src/buc/connection.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{
+	cnf::{GLOBAL_BUCKET, GLOBAL_BUCKET_ENFORCED},
+	err::Error,
+};
+use dashmap::DashMap;
+use std::sync::Arc;
+
+use super::store::{prefixed::PrefixedStore, Key, ObjectStore};
+
+// Helper type to represent how bucket connections are persisted
+pub type BucketConnections = DashMap<(String, String, String), Arc<dyn ObjectStore>>;
```

Comment:
What do the 3 strings here mean? This would probably be better to use a struct with named fields instead of a tuple.

Reply:
NS/DB/BU. I feel like a struct is overkill, but I can add a comment?

Reply:
I disagree, the tuple provides no information about the strings adding a struct with some fields with the right name will and will prevent accidental miss-ordering.